pkg/types/ccipocr3/common_types.go

```diff
@@ -4,18 +4,73 @@ import (
 	"encoding/hex"
 	"fmt"
 	"math/big"
+	"strings"
 )
 
+type Bytes []byte
+
+func NewBytesFromString(s string) (Bytes, error) {
+	if len(s) < 2 {
+		return nil, fmt.Errorf("Bytes must be of at least length 2 (i.e, '0x' prefix): %s", s)
+	}
+
+	if !strings.HasPrefix(s, "0x") {
+		return nil, fmt.Errorf("Bytes must start with '0x' prefix: %s", s)
+	}
+
+	b, err := hex.DecodeString(s[2:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode hex: %w", err)
+	}
+
+	return Bytes(b), nil
+}
+
+func (b Bytes) String() string {
+	return "0x" + hex.EncodeToString(b)
+}
+
+func (b Bytes) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"%s"`, b.String())), nil
+}
+
+func (b *Bytes) UnmarshalJSON(data []byte) error {
+	v := string(data)
+	if len(v) < 2 {
+		return fmt.Errorf("Bytes must be of at least length 2 (i.e, '0x' prefix): %s", v)
+	}
+
+	// trim the start and end double quotes
```

Comment:
q: do we need to handle edge-cases where there's leading / trailing whitespace?

Reply:
I only expect this to be used in tests, when we do actual chain reading we don't go from string -> bytes but read bytes directly and then deserialize appropriately.